### PR TITLE
HDDS-8994. [Multi-Tenant] Add CLI option to allow tenant creation on top of existing volumes

### PR DESCRIPTION
**Pull request by smengcl**
## What changes were proposed in this pull request?

Add `--force` option to tenant create CLI that allows a tenant to be created even when the volume exists. Note this option does NOT override any other failures, e.g. Ranger failures

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-8994

## How was this patch tested?

- [x] Added Ratis Tx UT.
- [x] Added CLI integration test case.

### CODE REVIEW

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java [hunk 1]
```diff
@@ -885,14 +885,21 @@ public void createTenant(String tenantId, TenantArgs tenantArgs)
...
+
     OmTenantArgs.Builder builder = OmTenantArgs.newBuilder();
     builder.setTenantId(tenantId);
     builder.setVolumeName(volumeName);
-    // TODO: Add more fields
-    // TODO: Include OmVolumeArgs in (Om)TenantArgs as well for volume creation?
+    builder.setForceCreationWhenVolumeExists(
+        tenantArgs.getForceCreationWhenVolumeExists());
+
+    // TODO: Add more fields. e.g. include OmVolumeArgs in (Om)TenantArgs
```
Comment (prashantpogde) [inline hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java:897]: do we need TODO ?

**Review by prashantpogde — COMMENTED** [review]
what happens when I use --force to create multiple tenants on the same volume ?

**Review by prashantpogde — APPROVED** [review]
Rest of the changes look good to me.

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java (hunk 1 above)
Comment (smengcl) [reply hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java:897]: Yes that is still a TODO item for volume creation customization during tenant creation. Unrelated to this item.

**Comment by smengcl** [comment]
> what happens when I use --force to create multiple tenants on the same volume ?

Good question.

Because of this Precondition check on refCount in OMTenantCreateRequest, it is not possible to assign multiple tenants to the same volume at the moment:

https://github.com/apache/ozone/blob/d3a7adf964da4df6d9cc9ed6df2c9c28ab3a44f7/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/tenant/OMTenantCreateRequest.java#L310-L311

Also, specifying a volume name different from the tenant name is not currently supported by the CLI. Arguably that is possible if someone put together a custom Java client that specifies a different volume name (ofc that user still need cluster admin / volume write permission to be able to do that since the permission check is on the OM side). -- This is already the case even without this PR with the current Tenant Create CLI. So I'd say it is not supported in official capacity right now because it is not fully tested.

**Comment by smengcl** [comment]
Thanks @prashantpogde for the review.

**Pull request merged**